src/kudu/gutil/ref_counted.cc

```diff
@@ -15,14 +15,14 @@ namespace subtle {
 
 RefCountedBase::RefCountedBase()
     : ref_count_(0)
-#ifndef NDEBUG
+#ifndef NDEBUG_DISABLED_FB
```

Comment:
in optimized mode of tp2 compilation, we use -DNDEBUG. In DEBUG mode, we don't
This would bring in the in_dtor_ extra member in DEBUG mode making it incompatible with OPT mode.
NDEBUG means "NO DEBUG"
NDEBUG_DISABLED_FB - would mean "NO DEBUG IS DISABLED IN FB"
To me seems like a double negative. Shouldn't it be named NDEBUG_FB?

Reply:
It's even worse than that! `#ifndef NDEBUG` is already a double negative: "enable if not debug is not defined" . So this would actually be a triple negative: "enable if not debug is disabled in FB is not defined" 😜

My intention is simply to disable this #ifdef without deleting the lines (to minimize the possibility of merge conflicts) so I thought appending _DISABLED_FB to the existing `#ifndef NDEBUG` lines would be a good way to do that.

Maybe I should just do:

```
#ifndef FB_DO_NOT_REMOVE // NDEBUG
```

What do you think?

Reply:
We can use FB_DO_NOT_REMOVE that way we have one define to find out all code that we have touched.
Its also unlikely that we will use the special checks or assertions around in_dtor_